adagrams/game.py

```diff
@@ -1,11 +1,122 @@
+import random
 def draw_letters():
-    pass
+    list_of_letters = []
+    LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+    }
+   
+    while len(list_of_letters) < 10:    
+        letter_chosen = random.choice(list(LETTER_POOL))
+        if LETTER_POOL[letter_chosen] > 0:
+            list_of_letters.append(letter_chosen)
+            LETTER_POOL[letter_chosen] -= 1
 
+    return list_of_letters
+        
 def uses_available_letters(word, letter_bank):
-    pass
+    #always make string 'word' be uppercase 
+    word = word.upper()
+    #generating a dictionary to keep track of how many each letter occurs in letter bank 
+    letter_frequency = {}
+    for letter in letter_bank:
+        if letter in letter_frequency:
+            letter_frequency[letter] += 1
+        else:
+            letter_frequency[letter] = 1    
+    #compare each letter in word to the key in letter frequency dictionary 
+    for elem in word: # loop through each letter in the word 
+        if elem in letter_frequency.keys(): 
+            if letter_frequency[elem] > 0:
+                letter_frequency[elem] -= 1
+            else: 
+                return False 
+        else:    
+            return False  
+    return True 
+
 
 def score_word(word):
-    pass
+    letter_score = {
+        "A" : 1,
+        "E" : 1, 
+        "I" : 1, 
+        "O" : 1,
+        "U" : 1, 
+        "L" : 1, 
+        "N" : 1, 
+        "R" : 1,
+        "S" : 1,
+        "T" : 1,
+        "D" : 2,
+        "G" : 2,
+        "B" : 3, 
+        "C" : 3, 
+        "M" : 3,
+        "P" : 3,
+        "F" : 4,
+        "H" : 4,
+        "V" : 4,
+        "W" : 4,
+        "Y" : 4,
+        "K" : 5,
+        "J" : 8, 
+        "X" : 8,
+        "Q" : 10, 
+        "Z" : 10
+    }
+    total_score = 0 
+    #turn every letter in the word to upper case 
```

Comment:
Comments are good, and sometimes you can have too much of a good thing! In most of this function, your comments are almost exactly the same as the lines of code they document. We avoid that in industry because the lines of code may change, but devs aren't always careful about changing the comments to match. It's also more for humans to read, but doesn't give them extra information.

What comments might you write to describe what this code is doing without describing how it does it (since that is what's most likely to change later)?